### PR TITLE
adapt the connection to the new beacons and make it more robust

### PR DESCRIPTION
**Pull request by KlaKalma**
This aims to adapt the actual beacon connection to the new schema : 
image

### CODE REVIEW

File: app/src/androidTest/java/ch/epfl/cs311/wanderwave/ui/TrackListScreenTest.kt
Comment (joriba) [inline app/src/androidTest/java/ch/epfl/cs311/wanderwave/ui/TrackListScreenTest.kt:1]: Remove commented out code

File: app/src/androidTest/java/ch/epfl/cs311/wanderwave/model/BeaconConnectionTest.kt
Comment (joriba) [inline app/src/androidTest/java/ch/epfl/cs311/wanderwave/model/BeaconConnectionTest.kt:1]: Also a lot of commented out code in this file.

**Review by joriba — CHANGES_REQUESTED** [review]
Hey, good work, looks fine.
I only see some commented out code, please remove that before we merge it in.

**Comment by joriba** [comment]
Also, we should start to look at the sonarCloud issues. There are 4 new of them, so maybe try to resolve (or conciously ignore) them.

**Review by mbouaissi — APPROVED** [review]
a part of the commented code, lgtm

**Pull request merged**